src/MaxMind/Db/Reader.php

```diff
@@ -237,8 +238,10 @@ private function findMetadataStart($filename)
         $fileSize = $fstat['size'];
         $marker = self::$METADATA_START_MARKER;
         $markerLength = self::$METADATA_START_MARKER_LENGTH;
+        $maxMetadataSize = self::$METADATA_MAX_SIZE;
+        // $maxMetadataSize = $fileSize - $markerLength;
 
-        for ($i = 0; $i < $fileSize - $markerLength + 1; $i++) {
+        for ($i = 0; $i < min($maxMetadataSize, $fileSize - $markerLength) + 1; $i++) {
```

Comment:
Yeah, I think you are right about the `+ 1`. I forgot that this is how many bytes from the end of the file where it starts looking back from.